adapters/appnexus/appnexus.go

```diff
@@ -95,7 +95,15 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.E
 	)
 
 	for i := 0; i < len(request.Imp); i++ {
+		// If the preprocessing failed, the server won't be able to bid on this Imp. Delete it, and note the error.
 		memberId, impAdPodId, err := preprocess(&request.Imp[i], defaultDisplayManagerVer)
+		if err != nil {
+			errs = append(errs, err)
+			request.Imp = append(request.Imp[:i], request.Imp[i+1:]...)
```

Comment:
I think we should refactor this more to make it simpler rather than modifying input request.Imp slice based on error. How about creating a new Imp slice and change the request.Imp at the end of the for loop.

Reply:
good call 👍🏼 

https://github.com/prebid/prebid-server/pull/2632/commits/5592ee5355b26355ac5565fd018328d4fba79db4 refactor append logic